### PR TITLE
feat: Migrate `head` content, meta data from MVP site

### LINKED ISSUE #4: Migrate meta data and static assets from MVP
Make sure the `<head>` meta tags and associated assets (icons, etc.) mirror those from the MVP site: https://github.com/USSF-ORBIT/ussf-portal-mvp/blob/main/_includes/head.html

### PR DESCRIPTION
**Pull request by suzubara**
## Description 

Adds all of the `head` elements, meta data, icons, etc. that are currently part of the MVP site (ref: https://github.com/USSF-ORBIT/ussf-portal-mvp/blob/main/_includes/head.html)

* Common & default `<head>` elements are defined in `pages/_app.tsx`. These can all be overridden by passing elements into `<Head>` on a page component (see the Test page example).

Fixes #4 

## Review Notes

* This introduces our first environment variable (see [NextJS docs](https://nextjs.org/docs/basic-features/environment-variables), it didn't seem to like looking for an `.envrc` file). If there is a better way to store the canonical domain per-environment I'm open to feedback. If not, I'll amend this PR to include docs about copying `.env.local.template` as part of dev setup.
* To test, visit the site locally and verify that the expected title & description for the home page and test page are visible in the browser window/source code. There's also a Cypress test to look for these.
* You should also verify none of the public assets result in a 404 error (such as the device-specific icons).
  * The favicon should show up in your browser window.
  * You can test the Safari tab icon by opening the site in Safari and opening another tab (see screenshot below).
  * You can also test iOS/android assets on a physical device by navigating to your computer's local IP at port 3000 when running the dev server. The icons should be used when saving the site to your home screen (see screenshots below).

image

IMG_6411
IMG_6412

### CODE REVIEW

**Comment by suzubara** [comment]
note to self: update documentation about `.env.local` before merging

File: cypress/integration/home.spec.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
     cy.contains('USSF Portal')
   })
+
+  it('contains the expected meta data', () => {
+    cy.document()
+    cy.get('head title').should('contain', 'Space Force Portal')
+    cy.get('head link[rel="canonical"]').should(
+      'have.attr',
+      'href',
+      'http://localhost:3000/'
```
Comment (abbyoung) [inline cypress/integration/home.spec.js:16]: flagging as question for eng sync today 🏳️

File: public/site.webmanifest [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "",
```
Comment (abbyoung) [inline public/site.webmanifest:2]: Should we include the name of the site here, or is this primarily for icons?

File: pages/index.tsx [hunk 3]
```diff
@@ -1,24 +1,25 @@
-import Head from 'next/head'
+// https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md#case-i-use-nextjs-and-im-getting-this-error-inside-of-links
```
Comment (abbyoung) [inline pages/index.tsx:1]: I really appreciate including this documentation along with the eslint-disable! What's the best way to track moving forward, to make sure that if/when the API is updated, we also update this code?

File: .env.local.template [hunk 4]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_SITE_URL='http://localhost:3000'
```
Comment (abbyoung) [inline .env.local.template:1]: Do we need to have this env variable to the Netlify deployment as well?

Comment (abbyoung) [reply .env.local.template:1]: (obvs different value but you know what I mean)

File: public/site.webmanifest (hunk 2 above)
Comment (abbyoung) [reply public/site.webmanifest:2]: fwiw it looks like whatever is set as the short_name is used when saving to an iPhone

File: .env.local.template (hunk 4 above)
Comment (abbyoung) [reply .env.local.template:1]: brain mush -- GOT IT.

File: pages/index.tsx (hunk 3 above)
Comment (abbyoung) [reply pages/index.tsx:1]: discussed in eng sync 🎉

File: public/site.webmanifest (hunk 2 above)
Comment (abbyoung) [reply public/site.webmanifest:2]: discussed in eng sync 🎉

**Review by abbyoung — DISMISSED** [review]
lgtm! thanks for talking through everything with me!

File: pages/_app.tsx [hunk 5]
```diff
@@ -1,8 +1,56 @@
 import 'styles/index.scss'
 import type { AppProps } from 'next/app'
+import Head from 'next/head'
```
Comment (esacteksab) [inline pages/_app.tsx:3]: You import `Head` from `'next/head'` but where is `'next/head'`?

Comment (suzubara) [reply pages/_app.tsx:3]: anything imported from `next/*` is from the nextJS package

**Review by esacteksab — APPROVED** [review]

**Review by esacteksab — APPROVED** [review]

**Review by abbyoung — APPROVED** [review]

**Pull request merged**